doc/source/file_formats.rst

```diff
@@ -1,23 +1,375 @@
+files. The metadata is given as a JSON style mapping of key/value
+pairs.
+
+#meta {"group": "Side chain bonds", "ifdef": "FLEXIBLE"}
+
+Metadata can also be added to a single line by adding an attribute
+statement as the last element.
+
+Atoms
+^^^^^
+
+Each line in the atoms section describes one particle, corresponding
+to a node in the molecular graph. The description comprises the atom
+name, atom type, residue identifier, mass and charge.
+
+Edges
+^^^^^
+
+Edges will be added to the molecular graph when required through
+interactions, but they can also be added explicitly by listing them
+under the edges subsections, specifying each edge to be added by the
+corresponding atom names.
+
+Interactions
+^^^^^^^^^^^^
+
+There are several options for subsections describing interactions
+between particles. Of these, bonds, angles, dihedrals, cmap, and
+constraints will automatically add the corresponding edges to the
+molecular graph, unless specified explicitly by setting an attribute
+'edge' to false in a subsection #meta or following a specific
+interaction.
+
+Each line in an interactions subsection specifies one interaction by
+listing the atoms involved by name, followed by the interaction
+parameters. For all interactions, the parameters are read as is and
+written to the output topology without interpreting and/or
+checking. Bond/constraint lengths, angles and dihedral angles may be
+used for generating missing coordinates.
+
+A full list of interactions is given below, corresponding to the list
+of intramolecular interactions available in Gromacs, with a number
+specifying the number of particles involved in the interaction. Note
+that improper dihedrals are listed as a separate interaction type,
+whereas in Gromacs these fall under the dihedrals section.
+
+* bonds(2)
+* angles(3)
+* dihedrals(4)
+* impropers(4)
+* constraints(2)
+* pairs(2)
+* pairs_nb(2)
+* SETTLE(1)
+* virtual_sites2(3)
+* virtual_sites3(4)
+* virtual_sites4(5)
+* position_restraints(1)
+* distance_restraints(2)
+* dihedral_restraints(4)
+* orientation_restraints(2)
+* angle_restraints(4)
+* angle_restraints_z(2)
+* cmap(...)
+
+Any of the subsections can be given multiply times, in which case they
+are additive. Do note that in the output topology specifying the same
+interaction several times (the same type and particles) will overrule
+any previous one, except when they are given different contexts (see
+below).
+
+Links
+~~~~~
+
+To generate a topology for a polymer or any molecule constructed from
+joining parts, Vermouth connects moleculetypes using links. A link
+describes how blocks are to be joined, what changes are effected in
+the atom lists and which interactions are added, removed, or
+altered. The changes in the atom and interaction lists are specified
+using the corresponding subsections as under moleculetype. Further
+subsections available for links are patterns, features, molmeta,
+edges, and non-edges
+
```

Comment:
From here on I would restructure the Links section somewhat:
- Links consist of atoms and edges. These are constructed from [atoms] [edges] and some interactions. 
- Links apply where the match the molecule being constructed.
- Atoms know something about (relative) resids (explain order attr here)
- Further filtering using patterns, features, molmeta, and non-edges.
- Atoms, edges and interactions; links can remove and modify atoms and interactions
- patterns, features, molmeta and non-edges

To me this flow makes more sense, because at the moment the order attribute falls from a blue sky in the Patterns subsection